saucerswap-faucetbot/main.js

```diff
@@ -0,0 +1,373 @@
+        replyThenDelete(`Quit your damn spamming! D:<`, userMsg);
+        break;
+      }
+
+      // add account to set of accountIds being processed
+      // remove accountId before every break statement
+      accountIdsBeingProcessed.add(accountId);
+
+      // check if accountId has token associated
+      const balanceCheckTx = await new AccountBalanceQuery()
+        .setAccountId(accountId)
+        .execute(client)
+        .catch((e) => {
+          const errorCode = e.status?._code;
+          let response;
+          if (errorCode == 15) {
+            response = `That account doesn't exist!`;
+            replyThenDelete(response, userMsg);
+          } else {
+            response = `Balance query on the account failed - try again?`;
+            replyThenDelete(response, userMsg);
+          }
+        });
+
+      // if there is an error, balanceCheckTx would be undefined - break in this case
+      if (!balanceCheckTx) {
+        accountIdsBeingProcessed.delete(accountId);
+        break;
+      }
+
+      if (!balanceCheckTx.tokens._map.get(tokenId)) {
+        replyThenDelete(
+          `In order to receive SAUCE, please have your account associated with its token ID: **${tokenId}**.
+          After doing so, please run the command again!`.replace(
+            /  +/g,
+            ""
+          ),
+          userMsg
+        );
+        accountIdsBeingProcessed.delete(accountId);
+        break;
+      }
+
+      // Query faucet_tracker table to see if accountId already pulled from the faucet
+      const result = await db.query(
+        `SELECT * FROM faucet_tracker WHERE accountId=?`,
+        [accountId]
+      );
+
+      if (result[0].length) {
+        replyThenDelete(
+          `Your account has pulled from the faucet already!\nThe next reset is in ${time.hrs} hours and ${time.mins} minutes!`,
+          userMsg
+        );
+        accountIdsBeingProcessed.delete(accountId);
+        break;
+      }
+
+      const payout = amount * Math.pow(10, decimals);
+      try {
+        const tx = await new TransferTransaction()
+          .addTokenTransfer(tokenId, treasuryId, -payout)
+          .addTokenTransfer(tokenId, accountId, payout)
+          .freezeWith(client)
+          .sign(treasuryKey);
+        const txResponse = await tx.execute(client);
+        const receipt = await txResponse.getReceipt(client);
+        // if tx success, insert values into table...
+        if (receipt.status._code == 22) {
+          await db.query( `INSERT INTO faucet_tracker (accountId) VALUE (?);`, [accountId] );
+          console.debug( `${payout / Math.pow(10, 6)} $SAUCE sent to ${accountId}` );
+          replyThenDelete(`Success!`, userMsg);
+        } else {
+          throw new Error(`Tx receipt status is not a success!`);
+        }
+      } catch (e) {
+        replyThenDelete(`Something went wrong! <@453749295309389834> HEEEEELP!!! <:aaa:921863627655508008>`, userMsg);
```

Comment:
user-config

Reply:
![image](https://user-images.githubusercontent.com/12617332/197709954-e19d9706-6e65-4fa4-b7f1-7af8f86fbd37.png)

That last part of the string is used to alert a developer via discord if the bot fails for whatever reason. I'm removing it if other devs want to customize it to their liking.